Sources/SmithyHTTPAPI/EndpointPropertyValue.swift

```diff
@@ -0,0 +1,62 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Foundation
+import Smithy
+import enum AwsCommonRuntimeKit.EndpointProperty
+
+public enum EndpointPropertyValue: Sendable, Equatable {
+    case bool(Bool)
+    case string(String)
+    indirect case array([EndpointPropertyValue])
+    indirect case dictionary([String: EndpointPropertyValue])
```

Comment:
Where are the possible endpoint property types defined?  Do we need an unknown case for this?